atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1862,12 +1881,44 @@ public void commit() {
     @Override
     public void commit(TransactionService transactionService) {
         try {
+            runPreCommitCallbacks();
             commitWithoutCallbacks(transactionService);
         } finally {
             close();
         }
     }
 
+    @com.palantir.lock.annotations.ReviewedRestrictedApiUsage
+    private void runPreCommitCallbacks() {
```

Comment:
I'm assuming you're tracking adding tests for this, once we have the full workflow working

Reply:
Yes